packages/editor/src/hooks/useTipTapEditor.ts

```diff
@@ -114,6 +115,13 @@ export const useTipTapEditor = (
       AttachmentNodeView(AttachmentRenderer),
     ],
     content,
+    onUpdate: ({ editor }) => {
+      const content = editor?.getHTML();
```

Comment:
Sur des gros documents, appeler `getHTML()` à chaque modification me fait un peu peur...
Surtout si `onContentChange` n'est pas défini, car dans ce cas cela ne sert à rien du tout !

Il vaudrait peut-être mieux ajouter un `useEffect()` dépendant de `[onContentChange]`, et faire un [late binding de l'event update](https://tiptap.dev/docs/editor/api/events#option-2-binding) ?

```suggestion
const updateCallback = useCallback( ({ editor }) => {
        onContentChange?.(editor?.getHTML());
        }, [onContentChange]
};

 useEffect(() => {
      onContentChange ? editor.on('update', updateCallback) : editor.off('update', updateCallback);
      }, [onContentChange]
```

Voire même ajouter un debounce...

Qu'en pensez-vous ?

Reply:
le debounce peut avoir une utilité mais je suis pas sûr que c'est à l'éditeur de faire ce debounce ou au niveau du composant parent

Reply:
A la réflexion, il vaut même mieux directement exposer directement `onUpdate` et laisser le "code client" faire ce qu'il veut : 

```suggestion
         onUpdate: onContentChange
```

en changeant le typage de  `onContentChange?: ({ editor })=>void;`

Reply:
@jcbe-ode Oui c'est vrai que appelé sans cesse le onupdate meme quand on ne l'utilise pas c'est pas fou j'avais pas pensé à ce cas là. Après oui je sais pas trop si ca peut causer des soucis d'appeler getHTML sur des gros contenu :/ 

En tout cas je vais faire tes modifs pour appeler le onContentChange que lorsqu'on en a besoin :)